runtime/util/hshelp.c

```diff
@@ -3229,6 +3232,9 @@ classIsModifiable(J9JavaVM * vm, J9Class * clazz)
 		rc = JNI_FALSE;
 	} else if ((J2SE_VERSION(vm) >= J2SE_19) && J9_ARE_ALL_BITS_SET(clazz->classFlags, J9ClassIsAnonymous)) {
 		rc = JNI_FALSE;
+	} else if (((classNameLength == 22) || (classNameLength == 23))
```

Comment:
length won't include the null terminator.  

Can this be written with an `else ` clause that has the name based matching in it?

ie:
```
} else {
 	/* Handle name based matches */
 	U_8* className = J9UTF8_DATA(J9ROMCLASS_CLASSNAME(clazz->romClass));
 	U_8 classNameLength = J9UTF8_LENGTH(className);
 	switch(classNameLength) {
 	case 22:
 	 	if (strncmp((char*)className, "com/ibm/jit/JITHelpers", 22) == 0) {
 	 	 	 rc = JNI_FALSE;
 	 	 }
 	 	break;
 	}
```

Reply:
updated